### PR TITLE
Add documentation

### PR DESCRIPTION
**Pull request by DevinHillenius**
## Description
Add documentation of Filesystem to filesystem repo.

## TODO:
- [x] Fix sidebar image ??
- [x] Add links to other documentations
- [x] Automatically set version

### CODE REVIEW

**Comment by libre-man** [comment]
The _build files (and probably others too) need to be added to the gitignore file. But looks nice!

File: docs/conf.py [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+
+# The version info for the project you're documenting, acts as replacement for
+# |version| and |release|, also used in various other places throughout the
+# built documents.
+#
+# The short X.Y version.
+# version = subprocess.check_output(
+#     ['git', 'describe', '--abbrev=0', '--tags']
+# ).decode('utf-8').strip()
+version = "v1.0.0"
```
Comment (libre-man) [inline docs/conf.py]: This should not be hard coded.

File: docs/plugins.rst [hunk 2]
```diff
@@ -0,0 +1,136 @@
+Editor Plug-ins
+=================
+
+Even though the CodeGrade filesystem works together with any editor by manually editing its special files,
+it is recommended to use it in combination with editor plugins.
+
+.. note:: The CodeGrade filesystem has to be installed and both the ``cgfs`` and the ``cgapi-consumer`` helper program must be available from the user's ``$PATH`` to successfully install and use the editor plugins.
+
+Atom Plug-in
```
Comment (libre-man) [inline docs/plugins.rst]: This should link to the plugins, not contain a copy of their documentation.

Comment (DevinHillenius) [reply docs/plugins.rst]: Fixed!

File: docs/conf.py (hunk 1 above)
Comment (DevinHillenius) [reply docs/conf.py]: Fixed!

File: codegra_fs/cgfs.py [hunk 3]
```diff
@@ -2017,6 +2017,117 @@ def check_version() -> None:
 
 def main() -> None:
     global cgapi
+    logger.info('Mounting... ')
```
Comment (libre-man) [inline codegra_fs/cgfs.py]: This shouldn't be added in this pull request.

Comment (DevinHillenius) [reply codegra_fs/cgfs.py]: Fixed!

**Pull request merged**